merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -219,34 +198,29 @@ export class Real {
   }
 }
 
-export class Discrete {
+export class Discrete<Schema> {
   public readonly __astNode: AST.DiscreteProfileExpression
+  private readonly __schemaInstance: Schema;
```

Comment:
If we never query it, I suspect we might not need to store it 🤔

Reply:
Yeah. I think as long as it has a type parameter, we don't need to actually supply a value and we get all the type checking benefits.